icaruscode/Timing/PMTBeamSignalNamespaceConverter_module.cc

```diff
@@ -0,0 +1,102 @@
+// Class:       PMTBeamSignalNameSpaceConverter_module
+// Plugin Type: producer (Unknown Unknown)
+// File:        PMTBeamSignalNameSpaceConverter_module.cc
+//
+// Generated at Thu Jun  5 14:59:32 2025 by Anna Heggestuen using cetskelgen
+// from cetlib version 3.18.02.
+////////////////////////////////////////////////////////////////////////
+#include "sbnobj/Common/PMT/Data/PMTBeamSignal.hh"
+#include "icaruscode/IcarusObj/Legacy/PMTBeamSignal.h"
+
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Principal/Run.h"
+#include "art/Framework/Principal/SubRun.h"
+#include "canvas/Utilities/InputTag.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+#include <memory>
+
+namespace sbn::timing {
+  class PMTBeamSignalNamespaceConverter;
+}
+/**
+ * @brief Copy PMTBeamSignal data product from icaruscode/IcarusObj namespace to sbnobj/Common
+ *
+ * This producer module simply copies what is stored in the icaruscode/IcarusObj data product 
+ * `std::vector<icarus::timing::PMTBeamSignal>` to a new sbnobj/Common data product 
+ * `std::vector<sbn::timing::PMTBeamSignal>`. This is to help with backward compatibility for
+ * files that are produced with the `icarus::timing::PMTBeamSignal` data product. Converting 
+ * the icarus::timing namespace to sbn::timing allows us to fill this variable in the CAFs. 
+ * 
+ */
+
+class sbn::timing::PMTBeamSignalNamespaceConverter : public art::EDProducer {
+public:
+  explicit PMTBeamSignalNamespaceConverter(fhicl::ParameterSet const& p);
+  // The compiler-generated destructor is fine for non-base
+  // classes without bare pointers or other resource use.
+
+  // Plugins should not be copied or assigned.
+  PMTBeamSignalNamespaceConverter(PMTBeamSignalNamespaceConverter const&) = delete;
+  PMTBeamSignalNamespaceConverter(PMTBeamSignalNamespaceConverter&&) = delete;
+  PMTBeamSignalNamespaceConverter& operator=(PMTBeamSignalNamespaceConverter const&) = delete;
+  PMTBeamSignalNamespaceConverter& operator=(PMTBeamSignalNamespaceConverter&&) = delete;
+
+  // Required functions.
+  void produce(art::Event& e) override;
+
+private:
+
+   /// RWM waveform instance label
+  art::InputTag const fRWMlabel;
+
+  // RWM times
+  std::vector<icarus::timing::PMTBeamSignal> fRWMTimes;
+  using BeamSignalCollection = std::vector<icarus::timing::PMTBeamSignal>;
+};
+
+
+sbn::timing::PMTBeamSignalNamespaceConverter::PMTBeamSignalNamespaceConverter(fhicl::ParameterSet const& p)
+  : EDProducer{p},  
+    fRWMlabel(p.get<art::InputTag>("RWMlabel"))
+{
+  // Call appropriate produces<>() functions here.
+  produces<std::vector<sbn::timing::PMTBeamSignal>>("RWM"); 
+
+  // Call appropriate consumes<>() for any products to be retrieved by this module.
+  consumes<std::vector<icarus::timing::PMTBeamSignal>>(fRWMlabel);
```

Comment:
Can you make this more general so that (if needed) we could also convert the `beamTiming:EW` products? For instance, I would rename `fRWMlabel` into a more generic `fSignalLabel` and replace the hardcoded "RWM" instance to copy the instance of the input label (`fSignalLabel.instance()` should do it).
```suggestion
    fSignalLabel(p.get<art::InputTag>("SignalLabel"))
{
  // Call appropriate produces<>() functions here.
  produces<std::vector<sbn::timing::PMTBeamSignal>>( fSignalLabel.instance()); 

  // Call appropriate consumes<>() for any products to be retrieved by this module.
  consumes<std::vector<icarus::timing::PMTBeamSignal>>(fSignalLabel);
```

Reply:
thanks for the suggestion! i made the namespace converter more general in [this commit ](https://github.com/SBNSoftware/icaruscode/pull/832/commits/2e7c592d852f75ee08a53c8c14053a26a5378151).